pg_search/src/postgres/customscan/pushdown.rs

```diff
@@ -134,33 +107,34 @@ pub unsafe fn try_pushdown_inner(
     root: *mut pg_sys::PlannerInfo,
     rti: pg_sys::Index,
     opexpr: OpExpr,
-    schema: &SearchIndexSchema
+    indexrel: &PgSearchRelation
 ) -> Option<Qual> {
     let args = opexpr.args();
-    let var = {
-        // inspect the left-hand-side of the operator expression...
-        let mut lhs = args.get_ptr(0)?;
-
-        while (*lhs).type_ == pg_sys::NodeTag::T_RelabelType {
-            // and keep following it as long as it's still a RelabelType
-            let relabel_type = lhs as *mut pg_sys::RelabelType;
-            lhs = (*relabel_type).arg as _;
-        }
-        nodecast!(Var, T_Var, lhs)?
-    };
+    let lhs = args.get_ptr(0)?;
     let rhs = args.get_ptr(1)?;
+    let pushdown = PushdownField::try_new(root, lhs, indexrel)?;
+    let search_field = pushdown.search_field();
 
-    let pushdown = PushdownField::try_new(root, var, schema)?;
-    let field = pushdown.search_field(schema)?;
-    if field.is_text() && !field.is_keyword() {
-        return None;
-    }
-
-    static EQUALITY_OPERATOR_LOOKUP: OnceLock<HashMap<pg_sys::Oid, &str>> = OnceLock::new();
-    match EQUALITY_OPERATOR_LOOKUP.get_or_init(|| initialize_equality_operator_lookup()).get(&opexpr.opno()) {
+    static EQUALITY_OPERATOR_LOOKUP: OnceLock<HashMap<PostgresOperatorOid, TantivyOperator>> = OnceLock::new();
+    match EQUALITY_OPERATOR_LOOKUP.get_or_init(|| unsafe { initialize_equality_operator_lookup(OperatorAccepts::All) }).get(&opexpr.opno()) {
```

Comment:
These `OnceLock` instances are still going to end up being initialized once per process: assuming they are actually worth memoizing, then I think that rather than having two of these maps, there should be a shared global static copy somewhere.